cunumeric/module.py

```diff
@@ -793,12 +832,12 @@ def diag(v, k=0):
         return v.diagonal(offset=k, axis1=0, axis2=1, extract=False)
     elif v.ndim == 2:
         return v.diagonal(offset=k, axis1=0, axis2=1, extract=True)
-    elif v.ndim > 2:
+    else:
```

Comment:
Otherwise mypy complains about a missing return for technically plausible negative value branch.